### ISSUE TITLE
test_lstsq in test_linalg.py is flaky on ROCm

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#53976]
Example failure snippet:

```
20:32:54 ======================================================================
20:32:54 ERROR [0.053s]: test_linalg_lstsq_cuda_float32 (__main__.TestLinalgCUDA)
20:32:54 ----------------------------------------------------------------------
20:32:54 Traceback (most recent call last):
20:32:54   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 899, in wrapper
20:32:54     method(*args, **kwargs)
20:32:54   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 292, in instantiated_test
20:32:54     raise rte
20:32:54   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 287, in instantiated_test
20:32:54     result = test_fn(self, *args)
20:32:54   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 572, in dep_fn
20:32:54     return fn(slf, device, *args, **kwargs)
20:32:54   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 572, in dep_fn
20:32:54     return fn(slf, device, *args, **kwargs)
20:32:54   File "test_linalg.py", line 210, in test_linalg_lstsq
20:32:54     a = random_well_conditioned_matrix(*shape, dtype=dtype, device=device)
20:32:54   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1601, in random_well_conditioned_matrix
20:32:54     u, _, v = x.svd()
20:32:54 RuntimeError: magma: The value of work_size(-9223372036854775808) is too large to fit into a magma_int_t (4 bytes)
```

cc @jianyuh @nikitaved @pearu @mruberry @heitorschueroff @walterddr @IvanYashchuk @jeffdaily @sunway513 @ROCmSupport @VitalyFedyunin

**Comment by jeffdaily** [comment]
This have anything to do with with the issue #53909 fixed recently?

**Comment by mruberry** [comment]
> This have anything to do with with the issue #53909 fixed recently?

No, all the failures occur before it.

**Comment by IvanYashchuk** [comment]
I think the issue and the fix are the same as for #53909, in that PR only CPU code was fixed. I'll submit fixes for the CUDA path that will probably resolve the ROCm failure.

**Comment by jeffdaily** [comment]
Doesn't seem like #54009 actually fixed this issue.  On the HUD (https://ezyang.github.io/pytorch-ci-hud/build/pytorch-master) the master commit where #54009 got merged actually had a failure.  See:

https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-linux-bionic-rocm4.0.1-py3.6-test1/5730//console

But it's not clear to me how this is not consistent behavior.  Instead it is flaky.

**Comment by IvanYashchuk** [comment]
It is strange that ROCm build is not in the list in GitHub UI
![image](https://user-images.githubusercontent.com/19621411/111761071-21ed1180-88a8-11eb-97d2-7f999d545665.png)

@mruberry This merge needs to be reverted https://github.com/pytorch/pytorch/commit/04a250609166327691105d463115ab16b6c3d420

**Comment by mruberry** [comment]
> It is strange that ROCm build is not in the list in GitHub UI
> ![image](https://user-images.githubusercontent.com/19621411/111761071-21ed1180-88a8-11eb-97d2-7f999d545665.png)
> 
> @mruberry This merge needs to be reverted [04a2506](https://github.com/pytorch/pytorch/commit/04a250609166327691105d463115ab16b6c3d420)

ROCm is typically run on PR CI. When we look at flakiness we're typically tracking the HUD: https://ezyang.github.io/pytorch-ci-hud/build/pytorch-master. Sure I can revert it, although the ROCm failure rate does appear lower.

**Comment by mruberry** [comment]
See also https://github.com/pytorch/pytorch/issues/54381.

Which tracks CUDA jit pinv and pinverse test failures.

**Comment by jeffdaily** [comment]
This was fixed by #54875 but did not automatically close this issue.  Closing manually.
